File: webserve/www/cgiFile/add.pl
#!/usr/bin/perl

use strict;
use warnings;
use CGI;

# Enable error output to the browser
use CGI::Carp qw(fatalsToBrowser);

my $cgi = CGI->new;

print $cgi->header('text/html');
print "<h1>Addition Results in perl</h1>";

# Retrieve the parameters
my $num1 = $cgi->param('num1');
my $num2 = $cgi->param('num2');

if (defined $num1 && defined $num2 && $num1 =~ /^\d+$/ && $num2 =~ /^\d+$/) {
    my $sum = $num1 + $num2;
    print "<output>$num1 + $num2 = $sum</output>";
} else {
    print "<output>Sorry, the script cannot turn your inputs into numbers (integers).</output>";
}

File: webserve/www/cgiFile/add.py
#!/usr/bin/env python3

import cgi, cgitb
cgitb.enable()

input_data = cgi.FieldStorage()


print("Content-Type: text/html") # HTML is following
print("")                        # Leave a blank line
print("<h1>Addition Results in python</h1>")
# print("input_data: ", input_data)
try:
    num1 = int(input_data.getvalue('num1'))
    num2 = int(input_data.getvalue('num2'))
except:
    print("<output>Sorry, the script cannot turn your inputs into numbers (integers).</output>")
    raise SystemExit(1)
print("<output>{0} + {1} = {2}</output>".format(num1, num2, num1 + num2))